### DISCUSSION TITLE
Issues with Whatsapp Channel using Whatsapp Cloud API

### DISCUSSION THREAD

**Discussion by sojan-official** [chatwoot/chatwoot#5092] [Self hosted]
Discussion tracking common issues in configuring WhatsApp via Whatsapp Cloud API

**Comment by bmishra5bestincity** [comment]
We are unable to send template message after adding a contact from contact tab.
While its working from conversation tab.

First Image not working, while last one working

![Screenshot 2022-08-20 122529](https://user-images.githubusercontent.com/13376723/185762123-5213fad2-6727-4192-9bf1-0e6bcdedd732.png)

![Screenshot 2022-08-20 122614](https://user-images.githubusercontent.com/13376723/185762127-4ede8ebf-60b7-4706-9b2a-685addca92aa.png)

**Reply by sojan-official** [reply]
@bmishra5bestincity seems like the template loader didn't work. Could you try refreshing the contact page and see if you are shown the template selector instead of the text area?

**Reply by pucilpet** [reply]
@bmishra5bestincity  This issue is being discussed here #5128

**Comment by dhwaj1902** [comment]
Hi, actually I have setup chatwoot recently, previously I was sending the template message through facebook graph API, but the message send through the graph API are not visible on chatwoot whatsapp. Can anyone please help me with this issue?

Or is there any API provided by the chatwoot?

**Reply by jain-jigar** [reply]
I have a similar issue. My WA template requires a document in the header. In the UI where you can select a template, there is no option to do so,

**Comment by Rohit-Mondal** [comment]
Hi, How can we get the newly added whatsapp templates. The new templetes are not getting fetched and I cant find any option to refresh it.

**Reply by voko3456** [reply]
+1

**Comment by Turki70** [comment]
Hi, can someone help me,

i made integrated CSML with Chatwoot and I made CSML the listener on my phone, but when the conversation ended, the customer still can receive messages from the agent, 

i need to close the session from the Chatwoot side.

**Comment by framatt** [comment]
How long does it take to update the available templates after creating them on Meta side?

**Comment by framatt** [comment]
I will answer myself here, 3 hours! So I made changes to the following file "/app/jobs/channels/whatsapp/templates_sync_scheduler_job.rb" and set it to 5 minutes.
